### PR TITLE
[14.0][MIG] base_name_search_improved

### PR DESCRIPTION
**Pull request by kittiu**
Standard migration, ready for review.

### CODE REVIEW

**Review by ps-tubtim — APPROVED** [review]
Code: Ok.
Functional Test: Ok.

**Review by Saran440 — APPROVED** [review]
Code no change :+1:

File: base_name_search_improved/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+The name search is the lookup feature to select a related record.
+For example, selecting a Customer on a new Sales order.
+
+For example, typing "john brown" doesn't match "John M. Brown".
+The relaxed search also looks up for records containing all the words,
+so "John M. Brown" would be a match.
+It also tolerates words in a different order, so searching
+for "brown john" also works.
+
+.. figure:: https://raw.githubusercontent.com/OCA/server-tools/11.0/base_name_search_improved/images/image0.png
```
Comment (newtratip) [inline base_name_search_improved/readme/DESCRIPTION.rst:14]:
```suggestion
.. figure:: ../images/image0.png
```

File: base_name_search_improved/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+Additionally, an Administrator can configure other fields to also lookup into.
+For example, Customers could be additionally searched by City or Phone number.
+
+.. figure:: https://raw.githubusercontent.com/OCA/server-tools/11.0/base_name_search_improved/images/image2.png
```
Comment (newtratip) [inline base_name_search_improved/readme/DESCRIPTION.rst:19]:
```suggestion
.. figure:: ../images/image2.png
```

File: base_name_search_improved/readme/CONFIGURE.rst [hunk 3]
```diff
@@ -0,0 +1,10 @@
+The fuzzy search is automatically enabled on all Models.
+Note that this only affects typing in related fields.
+The regular ``search()``, used in the top right search box, is not affected.
+
+Additional search fields can be configured at Settings > Technical > Database > Models,
+using the "Name Search Fields" field.
+
+.. figure:: https://raw.githubusercontent.com/OCA/server-tools/11.0/base_name_search_improved/images/image1.png
```
Comment (newtratip) [inline base_name_search_improved/readme/CONFIGURE.rst:8]:
```suggestion
.. figure:: ../images/image1.png
```

File: base_name_search_improved/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# © 2016 Daniel Reis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Improved Name Search",
+    "summary": "Friendlier search when typing in relation fields",
+    "version": "14.0.1.0.0",
+    "category": "Uncategorized",
+    "website": "https://github.com/OCA/server-tools",
+    "author": "Daniel Reis, Odoo Community Association (OCA), ADHOC SA",
```
Comment (newtratip) [inline base_name_search_improved/__manifest__.py:9]:
```suggestion
    "author": "Daniel Reis, ADHOC SA, Odoo Community Association (OCA)",
```

File: base_name_search_improved/views/ir_model_views.xml [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        </field>
+    </record>
+    <record id="view_model_form_new" model="ir.ui.view">
+        <field name="name">view.model.form</field>
+        <field name="model">ir.model</field>
+        <field name="arch" type="xml">
+            <form string="Custom Search">
+                <sheet>
+                    <div class="oe_button_box" name="buttons">
+                        <button
```
Comment (newtratip) [inline base_name_search_improved/views/ir_model_views.xml:28]: Could you check for toggle smart search button ?
Selection_373

**Comment by newtratip** [comment]
@kittiu, Please check codcov.

**Pull request merged**